src/test/java/org/javarosa/core/util/GeoDistanceTest.java

```diff
@@ -149,6 +153,90 @@ public void distance_isComputedForString() throws IOException, XFormParser.Parse
+                    bind("/data/point4").type("geopoint"),
+                    bind("/data/point5").type("geopoint"),
+                    bind("/data/distance").type("decimal").calculate("distance(concat(/data/point1, ';', /data/point2, ';', /data/point3, ';', /data/point4, ';', /data/point5))")
+                )),
+            body(
+                input("/data/point1")
+            )
+        ));
+
+        // http://www.mapdevelopers.com/area_finder.php?&points=%5B%5B38.253094215699576%2C21.756382658677467%5D%2C%5B38.25021274773806%2C21.756382658677467%5D%2C%5B38.25007793942195%2C21.763892843919166%5D%2C%5B38.25290886154963%2C21.763935759263404%5D%2C%5B38.25146813817506%2C21.758421137528785%5D%5D
+        assertThat(Double.parseDouble(scenario.answerOf("/data/distance").getDisplayText()),
+            IsCloseTo.closeTo(1801, 0.5));
+    }
+
+    @Test
+    public void distance_throwsForNonPoint() throws IOException, XFormParser.ParseException {
+        try {
+            Scenario.init("string distance", html(
+                head(
+                    title("String distance"),
+                    model(
+                        mainInstance(t("data id=\"string-distance\"",
+                            t("distance")
+                        )),
+                        bind("/data/distance").type("decimal").calculate("distance('foo')")
+                    )),
+                body(
+                    input("distance")
+                )
+            ));
+
+            fail("Exception expected");
+        } catch (RuntimeException e) {
+            assertThat(e.getCause(), instanceOf(XPathTypeMismatchException.class));
+            assertThat(e.getMessage(), containsString("The function 'distance' received a value that does not represent GPS coordinates"));
+        }
+    }
+
+    @Test
+    public void distance_isComputedForMultipleArguments() throws IOException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("string distance", html(
+            head(
+                title("Multi parameter distance"),
+                model(
+                    mainInstance(t("data id=\"string-distance\"",
+                        t("point1", "38.253094215699576 21.756382658677467 0 0"),
+                        t("point3", "38.25007793942195 21.763892843919166 0 0"),
+                        t("point4", "38.25290886154963 21.763935759263404 0 0"),
+                        t("point5", "38.25146813817506 21.758421137528785 0 0"),
+                        t("distance")
+                    )),
+                    bind("/data/point1").type("geopoint"),
+                    bind("/data/point3").type("geopoint"),
+                    bind("/data/point4").type("geopoint"),
+                    bind("/data/point5").type("geopoint"),
+                    bind("/data/distance").type("decimal").calculate("distance(/data/point1, '38.25021274773806 21.756382658677467 0 0', /data/point3, /data/point4, /data/point5)")
```

Comment:
I'm guessing "point 2" is not pulled from a node to test that literals work fine here? I think if that's a concern it should move out to a separate test (or maybe a unit test of `XPathFuncExpr`) so it's not hidden in here.